genesyscloud/resource_genesyscloud_tf_export.go

```diff
@@ -0,0 +1,594 @@
+		return diagErr
+	}
+
+	tfstate := terraform.NewState()
+	for _, resource := range resources {
+		resourceState := &terraform.ResourceState{
+			Type:     resource.Type,
+			Primary:  resource.State,
+			Provider: "provider.genesyscloud",
+		}
+		tfstate.RootModule().Resources[resource.Type+"."+resource.Name] = resourceState
+	}
+
+	data, err := json.MarshalIndent(tfstate, "", "  ")
+	if err != nil {
+		return diag.Errorf("Failed to encode state as JSON: %v", err)
+	}
+
+	log.Printf("Writing export state file to %s", stateFilePath)
+	if err := writeToFile(data, stateFilePath); err != nil {
+		return err
+	}
+
+	// This outputs terraform state v3, and there is currently no public lib to generate v4 which is required for terraform 0.13+.
+	// However, the state can be upgraded automatically by calling the terraform CLI. If this fails, just print a warning indicating
+	// that the state likely needs to be upgraded manually.
+	cliError := `Failed to run the terraform CLI to upgrade the generated state file. 
+	The generated tfstate file will need to be upgraded manually by running the 
+	following in the state file's directory:
+	'terraform state replace-provider registry.terraform.io/-/genesyscloud registry.terraform.io/mypurecloud/genesyscloud'`
+
+	tfpath, err := exec.LookPath("terraform")
+	if err != nil {
+		log.Println(cliError)
+		return nil
+	}
+
+	// exec.CommandContext does not auto-resolve symlinks
+	fileInfo, err := os.Lstat(tfpath)
+	if err != nil {
+		log.Println(cliError)
+		return nil
+	}
+	if fileInfo.Mode()&os.ModeSymlink != 0 {
+		tfpath, err = os.Readlink(tfpath)
+		if err != nil {
+			log.Println(cliError)
+			return nil
+		}
+	}
+
+	cmd := exec.CommandContext(ctx, tfpath)
+	cmd.Args = append(cmd.Args, []string{
+		"state",
+		"replace-provider",
+		"-auto-approve",
+		"-state=" + stateFilePath,
+		"registry.terraform.io/-/genesyscloud",
+		providerSource,
+	}...)
+
+	log.Printf("Running 'terraform state replace-provider' on %s", stateFilePath)
+	if err := cmd.Run(); err != nil {
+		log.Println(cliError)
+		return nil
+	}
+	return nil
+}
+
+func writeJSONConfig(jsonMap map[string]interface{}, path string) diag.Diagnostics {
+	dataJSONBytes, err := json.MarshalIndent(jsonMap, "", "  ")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("Writing export config file to %s", path)
+	if err := writeToFile(dataJSONBytes, path); err != nil {
+		return err
+	}
+	return nil
+}
+
+func sanitizeConfigMap(
+	resourceType string,
+	configMap map[string]interface{},
+	prevAttr string,
+	exporters map[string]*ResourceExporter,
+	exportingState bool) bool {
+
+	exporter := exporters[resourceType]
+	for key, val := range configMap {
+		currAttr := key
+		wildcardAttr := "*"
+		if prevAttr != "" {
+			currAttr = prevAttr + "." + key
+			wildcardAttr = prevAttr + "." + "*"
+		}
+
+		if currAttr == "id" {
+			// Strip off IDs from the root resource
+			delete(configMap, currAttr)
+			continue
+		}
+
+		switch val.(type) {
```

Comment:
It might be worth adding a default entry for unrecognized values, if you want to emit an warning or fail.

Reply:
There can be ints and floats, there was just nothing to do with them here.